thruster/src/core/errors.rs

```diff
@@ -26,38 +25,42 @@ pub trait ErrorSet<C> {
 }
 
 impl<C: Context> ErrorSet<C> for ThrusterError<C> {
-    fn parsing_error(context: C, error: &str) -> ThrusterError<C> {
+    fn parsing_error(mut context: C, error: &str) -> ThrusterError<C> {
+        context.status(400);
+
         ThrusterError {
             context,
             message: format!("Failed to parse '{}'", error),
-            status: 400,
             cause: None,
         }
     }
 
-    fn generic_error(context: C) -> ThrusterError<C> {
+    fn generic_error(mut context: C) -> ThrusterError<C> {
```

Comment:
Maybe some juniors can think this error can be used everywhere, as the name suggests that it's a "catch-all" error. What about 

- renaming `generic_error` to `request_error` to suggest parsing query, header, or path param can cause it. Returns status 400.
- create a `process_error` that can be used during matches, guards, or other normal operation errors. Return status 400.
- create a new `server_error` that covers the case when the server has to panic or external services are failing. Returns status 500.

Reply:
I'll take a look at this! Originally these errors were meant to be more useful as default cases and the developer expected to implement their own errors to handle them better.

I'm thinking an appropriate piece of documentation describing that intent as well a perhaps adding aliases for the errors you've described above.